routes/api.js

```diff
@@ -90,4 +94,32 @@ api.delete('/friendships/user/:secret', async function (req, res) {
     })
 })
 
+// Route for merging accounts
+api.put('/user/:secret/merge', async function (req, res) {
+  const { secret } = req.params
+  const { mergeUrl } = req.body
+  const mergeSecret = mergeUrl.split('/user/')[1]
+  // Checks for invalid url and prevents merging into itself
+  if (mergeSecret === undefined || secret === mergeSecret) {
+    return res.status(400).send('Bad url')
+  }
+
+  // Checks for existing account
+  const userIds = await UserService.getUserIdsFromSecrets([mergeSecret, secret])
+  const mergeEmail = await UserService.getUserEmailBySecret(mergeSecret)
+  if (userIds[mergeSecret] === null || userIds[secret] === null || Object.keys(userIds).length !== 2) {
+    return res.status(404).send()
+  }
```

Comment:
These are both data-level calls, not service-level - rather than working at a the user abstraction layer (checking that their account is merged), they are at the data layer (checking the merge layer). This logic would be better encapsulated in a UserService function, rather than requiring the route to understand what the data model looks like.

Indeed, this validation can be moved to `UserMergeService.mergeAccounts`, allowing us to reuse the existing error handling structure down below.